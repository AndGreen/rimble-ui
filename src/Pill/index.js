import React, { Component } from "react";
import styled from "styled-components";
import { themeGet } from "styled-system";
import { tint } from "polished";

import Box from "../Box";

const Pill = styled(Box).attrs({
  tintLevel: props => (props.selected ? "0.75" : "0.88")
})`
  & {
    cursor: pointer;
    border: ${props => (props.selected ? "1px solid" : "none")};
    border-color: ${props => (props.selected ? props.color : "none")};
    background-color: ${props =>
      tint(
        props.tintLevel,
        themeGet("colors." + props.color, props.color)(props)
      )};
  }
  &:hover {
    border-color: transparent;
    background-color: ${props =>
      tint(0.8, themeGet("colors." + props.color, props.color)(props))};
  }
`;

Pill.defaultProps = {
  color: "dark-gray",
  px: 3,
  py: 1,
  fontSize: 1,
  height: 2,
  lineHeight: "solid",
  borderRadius: 3,
  display: "inline-flex",
  alignItems: "center",
  selected: false
};

Pill.displayName = "Pill";

export default Pill;
