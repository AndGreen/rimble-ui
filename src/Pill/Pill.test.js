import Pill from "./";

describe("Pill component sanity", () => {
  it("has name", () => {
    expect(Pill.displayName).toBe("Pill");
  });

  it("matches default snapshot", () => {
    const component = render(<Pill color={"blue"} />);
    expect(component).toMatchSnapshot();
  });
});
