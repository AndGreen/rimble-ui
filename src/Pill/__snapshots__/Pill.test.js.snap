// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill component sanity matches default snapshot 1`] = `
.c0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: blue;
  font-size: 14px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 2px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
}

.c0 {
  box-sizing: border-box;
  text-align: left;
}

.c0 {
  cursor: pointer;
  border: none;
  border-color: none;
  background-color: #e0e0ff;
}

.c0:hover {
  border-color: transparent;
  background-color: #ccf;
}

<div
  class="c0"
  color="blue"
  display="inline-flex"
  font-size="1"
  height="2"
  tintlevel="0.88"
>
  
</div>
`;
